test/m3u8.test.js

```diff
@@ -803,6 +827,132 @@ QUnit.test('can be constructed', function(assert) {
   assert.notStrictEqual(typeof new Parser(), 'undefined', 'parser is defined');
 });
 
+QUnit.test('can set custom parsers', function(assert) {
+  const parser = new Parser();
+  const manifest = [
+    '#EXTM3U',
+    '#EXT-X-VERSION:3',
+    '#EXT-X-TARGETDURATION:10',
+    '#EXT-X-MEDIA-SEQUENCE:0',
+    '#EXT-X-PROGRAM-DATE-TIME:2017-07-31T20:35:35.053+00:00',
+    '#VOD-STARTTIMESTAMP:1501533337573',
+    '#VOD-TOTALDELETEDDURATION:0.0',
+    '#VOD-FRAMERATE:29.97',
+    ''
+  ].join('\n');
+
+  parser.addParser({
+    expression: /^#VOD-STARTTIMESTAMP/,
+    customType: 'startTimestamp',
+    isManifestLevel: true
+  });
+  parser.addParser({
+    expression: /^#VOD-TOTALDELETEDDURATION/,
+    customType: 'totalDeleteDuration',
+    isManifestLevel: true
+  });
+  parser.addParser({
+    expression: /^#VOD-FRAMERATE/,
+    customType: 'framerate',
+    dataParser: (line) => (line.split(':')[1]),
+    isManifestLevel: true
+  });
+
+  parser.push(manifest);
+  assert.strictEqual(
+    parser.manifest.custom.startTimestamp,
+    '#VOD-STARTTIMESTAMP:1501533337573',
+    'sets custom timestamp line'
+  );
+
+  assert.strictEqual(
+    parser.manifest.custom.totalDeleteDuration,
+    '#VOD-TOTALDELETEDDURATION:0.0',
+    'sets custom delete duration'
+  );
+
+  assert.strictEqual(parser.manifest.custom.framerate, '29.97', 'sets framerate');
+});
+
+QUnit.test('segment level custom data', function(assert) {
+  const parser = new Parser();
+
+  const manifest = [
+    '#EXTM3U',
+    '#VOD-TIMING:1511816599485',
+    '#COMMENT',
+    '#EXTINF:8.0,',
+    'ex1.ts',
+    '#VOD-TIMING',
+    '#EXTINF:8.0,',
+    'ex2.ts',
+    '#VOD-TIMING:1511816615485',
+    '#EXT-UNKNOWN',
+    '#EXTINF:8.0,',
+    'ex3.ts',
+    '#VOD-TIMING:1511816623485',
+    '#EXTINF:8.0,',
+    'ex3.ts',
+    '#EXT-X-ENDLIST'
+  ].join('\n');
+
+  parser.addParser({
+    expression: /^#VOD-TIMING/,
+    customType: 'vodTiming'
+  });
+
+  parser.push(manifest);
+  assert.equal(
+    parser.manifest.segments[0].custom.vodTiming,
+    '#VOD-TIMING:1511816599485',
+    'parser attached segment level custom data'
+  );
+  assert.equal(
+    parser.manifest.segments[1].custom.vodTiming,
+    '#VOD-TIMING',
+    'parser got segment level custom data without :'
+  );
+});
+
+QUnit.test('playlist level custom data', function(assert) {
+  const parser = new Parser();
+
+  const manifest = [
+    '#EXTM3U',
+    '#EXT-X-VERSION:6',
+    '#PLAYLIST-LABEL:MED',
```

Comment:
I'm not sure we should have "Playlist Level" tags like this. Do you have a valid use case you need this for, or was this added for completeness? There isn't really any notion of media playlist tags within the master manifest as attribute lists are used for any playlist info. It makes more sense to me that all tags in a master manifest are "Manifest Level" tags, and anything that would be considered a "Playlist Level" tag should be a "Manifest Level" tag within the media playlist itself. Because of this, I also lean more towards having "Manifest Level" be the default and require an option to specify that a parser is for a "Media Segment" tag. Thoughts?

Reply:
This is mostly for completeness. Technically this would be supported regardless of whether we signal it or not since the parser doesn't differentiate between segments and playlists at that level. I agree that the more logical case is that playlist tags belong at the 'manifest level' of the specific playlist. As to what the default should be, I'm pretty happy either way. Garett and I discussed `isManifestLevel` as a way to avoid having to name something that applies equally to segments and playlists, given that both manifest-level and segment-level seem to be common use cases. If we don't think that openly supporting this style of playlist tags is a good idea, it should be really simple for us to revert the changes.

Reply:
I think that we should not support these types of playlist tags, but instead have the idea of Segment level tags and Manifest level tags. Whichever is the default is not too important as long as we remove the confusion of playlist vs manifest level, so keeping the `isManifestLevel` option is fine.

Reply:
I think @mjneil is right. Originally I recalled seeing "playlist" tags within the master manifest where they existed above their relevant playlists, similar to the segment style, however, that seems like a bad use case. Playlist data should either be in the comma delineated `#EXT-X-STREAM-INF` for the relevant playlist in the master manifest (that data we already parse, custom or not), or, as discussed here, the custom data should be at the top level of the appropriate playlist as a "manifest level" tag. If someone tries the other method, I think not supporting it and instead directing them to one of these two methods would be optimal.

Reply:
I'll swap the name of the param back to `segment` and remove the tests regarding the master playlist functionality